guppylang/prelude/builtins.py

```diff
@@ -722,6 +725,18 @@ def reverse(self: linst[L]) -> linst[L]: ...
     def sort(self: list[T]) -> None: ...
 
 
+n = guppy.nat_var(builtins, "n")
+
+
+@guppy.extend_type(builtins, array_type_def)
+class Array:
+    @guppy.hugr_op(builtins, DummyOp("ArrayGet"))
+    def __getitem__(self: array[T, n], idx: int) -> T: ...
```

Comment:
The type varable `T` is declared as non-linear, so it cannot be instantiated with `qubit` (see error tests).

We first need some changes to the type checker in order to fully support linear arrays